app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -1484,12 +1479,7 @@ class BrowserTabViewModel @Inject constructor(
         sitePermissionsAllowedToAsk: Array<String>,
     ) {
         viewModelScope.launch(dispatchers.io()) {
-            val url = request.origin.toString()
-            val sitePermissionsGranted = sitePermissionsManager.getSitePermissionsGranted(url, tabId, sitePermissionsAllowedToAsk)
-            val sitePermissionsToAsk = sitePermissionsAllowedToAsk.filter { !sitePermissionsGranted.contains(it) }.toTypedArray()
-            if (sitePermissionsGranted.isNotEmpty()) {
-                command.postValue(GrantSitePermissionRequest(sitePermissionsGranted, request))
-            }
+            val sitePermissionsToAsk = sitePermissionsManager.getSitePermissionsForUserToHandle(tabId, request)
```

Comment:
This is not needed as we already pass it as an argument: `sitePermissionsAllowedToAsk`. Therefore, `sitePermissionsManager` doesn't need to be injected in the `BrowserTabViewModel` anymore.
Also the following `if()` it's not needed either because it is checked in the `WebViewClientListener` just before calling this function

Reply:
Done